[One Workflow] Improve run workflow modal

VladimirFilonov opened the pull request:
## Summary

feat(workflows): add index-based test input and enhance alert execution UI
Added workflow_execute_index_form.tsx to allow selecting a document from an index as test input
Enhanced workflow_execute_event_form.tsx with time range picker, query filtering, selection-based event payload, and error handling
Updated workflow_execute_modal.tsx to integrate new execution forms and flow
Improved translations with trigger descriptions and labels
Adjusted workflow detail header/page to support new execution options
Cleaned up plugin manifest (kibana.jsonc) and removed obsolete test_workflow_modal.tsx
Updated shared types for new execution inputs and interactions
Refs: workflows management manual triggers and run workflow UX improvements

<img width="1227" height="690" alt="image" src="https://github.com/user-attachments/assets/c4287daf-8945-4d6a-a6e6-979a0a38d83b" />

<img width="1216" height="678" alt="image" src="https://github.com/user-attachments/assets/1775afb6-226a-48df-9177-d98597d41769" />


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

VladimirFilonov committed:
feat(workflows): add index-based test input and enhance alert execution UI
Added workflow_execute_index_form.tsx to allow selecting a document from an index as test input
Enhanced workflow_execute_event_form.tsx with time range picker, query filtering, selection-based event payload, and error handling
Updated workflow_execute_modal.tsx to integrate new execution forms and flow
Improved translations with trigger descriptions and labels
Adjusted workflow detail header/page to support new execution options
Cleaned up plugin manifest (kibana.jsonc) and removed obsolete test_workflow_modal.tsx
Updated shared types for new execution inputs and interactions
Refs: workflows management manual triggers and run workflow UX improvements

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

VladimirFilonov committed:
fix(translations): correct typo in alert trigger description and update time range state management

- Fixed typo from "exited" to "existing" in the alert trigger description.
- Corrected state variable name from `timeRage` to `timeRange` in `workflow_execute_event_form.tsx` for consistency.
- Removed unused `handleTestClick` function from workflow detail header stories.

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/342959)
* Commit: 5809d6fe56714f3b07f7799cd06055e43285725a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236173)

### Failed CI Steps
* [FTR Configs #19](https://buildkite.com/elastic/kibana-pull-request/builds/342959#01997ac8-f923-44b0-a3c5-aa52c73c3eef)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342959#01997ac8-f923-44b0-a3c5-aa52c73c3eef) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342959/jobs/01997ac8-f923-44b0-a3c5-aa52c73c3eef/artifacts/01997af4-c486-4c03-a13e-d031d5282a5b) FTR Configs #<span></span>19 / Space awareness outputs With Fleet server setup in a specific space "after all" hook for "should allow to create a default logstash output"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/759649f3a36f97812a6e60a91d440c4aac569cfa) | [after](https://github.com/elastic/kibana/commit/5809d6fe56714f3b07f7799cd06055e43285725a) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 1.7MB | 1.7MB | +2.8KB |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/759649f3a36f97812a6e60a91d440c4aac569cfa) | [after](https://github.com/elastic/kibana/commit/5809d6fe56714f3b07f7799cd06055e43285725a) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 19 | 17 | -2 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/759649f3a36f97812a6e60a91d440c4aac569cfa) | [after](https://github.com/elastic/kibana/commit/5809d6fe56714f3b07f7799cd06055e43285725a) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 0 | 2 | +2 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/759649f3a36f97812a6e60a91d440c4aac569cfa) | [after](https://github.com/elastic/kibana/commit/5809d6fe56714f3b07f7799cd06055e43285725a) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 19 | 17 | -2 |


</details>


### History
* :yellow_heart: [Build #342927](https://buildkite.com/elastic/kibana-pull-request/builds/342927) was flaky 4ce7542d8397d565f3f4ba7c614f446817ef7d0e
* :broken_heart: [Build #342844](https://buildkite.com/elastic/kibana-pull-request/builds/342844) failed eb0b7453e723afc887fb7147908bc1bf02bce239

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342959","number":342959,"commit":"5809d6fe56714f3b07f7799cd06055e43285725a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342927","number":342927,"commit":"4ce7542d8397d565f3f4ba7c614f446817ef7d0e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342844","number":342844,"commit":"eb0b7453e723afc887fb7147908bc1bf02bce239"}],"number":342959}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

semd commented on src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx:
```diff
@@ -0,0 +1,407 @@
+  errors,
+  setErrors,
+}: WorkflowExecuteEventFormProps): React.JSX.Element => {
+  const { services } = useKibana<CoreStart & WorkflowsPluginStartDependencies>();
```
We can now load the `useKibana` from here:

https://github.com/elastic/kibana/blob/dec7fe494895438246e8ce8082004390f07f1edd/src/platform/plugins/shared/workflows_management/public/hooks/use_kibana.ts#L17

So we don't have to worry about adding the correct types. It's already typed with all the services we have in the workflows plugin. 
It's also easier to mock in the tests

semd commented on src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx:
```diff
@@ -0,0 +1,407 @@
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFormRow label="Documents" fullWidth>
+            <EuiFieldText
+              placeholder="Filter your data using KQL syntax"
+              value={queryString}
+              onChange={(e) => setQueryString(e.target.value)}
+              onKeyPress={(e) => {
+                if (e.key === 'Enter') {
+                  fetchDocuments();
+                }
+              }}
+              fullWidth
+              prepend={<EuiIcon type="search" />}
+              data-test-subj="workflow-query-input"
+            />
```
For the search bar, I think it would be nice to try reusing the `SearchBar` component. 

To get it:

```
const { unifiedSearch } = useKibana().services.unifiedSearch;
const { SearchBar } = unifiedSearch.ui;
```

Example here:
https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/common/components/search_bar/index.tsx

<img width="1384" height="119" alt="Captura de pantalla 2025-09-26 a les 13 11 25" src="https://github.com/user-attachments/assets/ff559a74-61f8-4462-8588-eca5dfd19d76" />

And maybe it would make sense to add a timerange input as well to the form (maybe later)

VladimirFilonov committed:
Implement unflattenObject utility and process all alerts' fields for workflows

- Added a new utility function `unflattenObject` to transform flat objects into nested structures.
- Updated the `updateEventData` function in `WorkflowExecuteEventForm` to utilize the new utility for processing alert data.
- Simplified the `handleRunClick` function in `WorkflowDetailPage` to always open the workflow execution modal, removing unnecessary input checks.

VladimirFilonov committed:
Add unifiedSearch plugin to workflows management and enhance error display in forms

- Included the `unifiedSearch` plugin in the Kibana workflows management configuration.
- Updated the `EuiCallOut` components in `workflow_execute_event_form.tsx` and `workflow_execute_index_form.tsx` to include the `announceOnMount` prop for better accessibility.
- Replaced the `EuiComboBox` with `DataViewPicker` in `workflow_execute_index_form.tsx` for improved data view selection experience.

VladimirFilonov committed:
Enhance workflows management with unifiedSearch integration and UI improvements

- Added `UnifiedSearchPublicPluginStart` to workflows management dependencies.
- Updated `workflow_execute_index_form.tsx` to utilize the `SearchBar` from the unified search plugin for improved query input.
- Wrapped `DataViewPicker` in an `EuiPanel` for better UI presentation and adjusted layout spacing for a cleaner look.

VladimirFilonov committed:
Update workflows management to improve query handling and UI elements

- Added `zIndex` to the `ResizableSuggestionsListDiv` for better layering in the suggestions component.
- Included the `data` plugin in the workflows management configuration for enhanced functionality.
- Refactored `workflow_execute_index_form.tsx` to utilize the `buildEsQuery` function for constructing search queries, improving query management and data fetching logic.

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

VladimirFilonov committed:
Refactor workflow forms to use custom `useKibana` hook

- Updated `workflow_execute_event_form.tsx` and `workflow_execute_index_form.tsx` to import `useKibana` from a local hook instead of the Kibana React plugin.
- Simplified service access in both components by removing unnecessary type parameters.

VladimirFilonov committed:
Refactor WorkflowExecuteEventForm to integrate unified search components

- Removed the custom TimeRangePicker component and replaced it with the SearchBar from the unified search plugin for improved query handling.
- Updated time range state management to align with the new TimeRange type.
- Simplified query handling by utilizing the Query type for better type safety and clarity.
- Enhanced UI elements for a more cohesive user experience in the workflow execution form.

VladimirFilonov committed:
Enhance workflows management with new dependencies and improved query handling

- Added `DataPublicPluginStart` and `SpacesPluginStart` to the workflows management dependencies for enhanced functionality.
- Updated `WorkflowExecuteEventForm` to retrieve the active space ID and integrate it into alert fetching logic.
- Refactored query handling in `WorkflowExecuteEventForm` to utilize `buildEsQuery` for better query construction.
- Improved `WorkflowExecuteIndexForm` to manage time range state and integrate it into document fetching logic, enhancing overall user experience.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346136)
* Commit: 8eb8d7b8f06d132d3a5bdf6255e1adb873f10584

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/346136#01999ec4-743f-4eea-b4bc-2ad5e54b36f6)

### History
* :broken_heart: [Build #345562](https://buildkite.com/elastic/kibana-pull-request/builds/345562) failed a3239b6f06d0016ff6c2bea1d5abf821a5f1939e
* :yellow_heart: [Build #342959](https://buildkite.com/elastic/kibana-pull-request/builds/342959) was flaky 5809d6fe56714f3b07f7799cd06055e43285725a
* :yellow_heart: [Build #342927](https://buildkite.com/elastic/kibana-pull-request/builds/342927) was flaky 4ce7542d8397d565f3f4ba7c614f446817ef7d0e
* :broken_heart: [Build #342844](https://buildkite.com/elastic/kibana-pull-request/builds/342844) failed eb0b7453e723afc887fb7147908bc1bf02bce239

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346136","number":346136,"commit":"8eb8d7b8f06d132d3a5bdf6255e1adb873f10584"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345562","number":345562,"commit":"a3239b6f06d0016ff6c2bea1d5abf821a5f1939e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342959","number":342959,"commit":"5809d6fe56714f3b07f7799cd06055e43285725a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342927","number":342927,"commit":"4ce7542d8397d565f3f4ba7c614f446817ef7d0e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342844","number":342844,"commit":"eb0b7453e723afc887fb7147908bc1bf02bce239"}],"number":346136}
buildkite-pr-comment-->

VladimirFilonov committed:
Refactor Workflow Execute Modal and Suggestions List

- Removed zIndex from ResizableSuggestionsListDiv for improved layout handling.
- Updated WorkflowExecuteModal to include Global styles for z-index management.
- Simplified handleRunClick logic in WorkflowDetailPage to streamline modal opening.
- Removed unused handleTestClick from storybook examples for cleaner code.

These changes enhance the UI consistency and improve the overall user experience in the workflows management section.

VladimirFilonov committed:
Remove unused handleTestClick prop from WorkflowDetailHeader test for cleaner code.

VladimirFilonov committed:
Add data view, field formats, unified search, data, and spaces mocks to workflows management

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346548)
* Commit: 73f6e0027a76593fe8f77fb9ce2adb16e3e94761

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/346548#0199a3c6-0ee2-4971-a0af-538bf5edaad1)

### History
* :broken_heart: [Build #346379](https://buildkite.com/elastic/kibana-pull-request/builds/346379) failed ca92182ba2d876ceace2de8ea55e3d5d5d106efc
* :broken_heart: [Build #346335](https://buildkite.com/elastic/kibana-pull-request/builds/346335) failed 55265594a899376fa99d738cc53c36bef56da794
* :broken_heart: [Build #346136](https://buildkite.com/elastic/kibana-pull-request/builds/346136) failed 8eb8d7b8f06d132d3a5bdf6255e1adb873f10584
* :broken_heart: [Build #345562](https://buildkite.com/elastic/kibana-pull-request/builds/345562) failed a3239b6f06d0016ff6c2bea1d5abf821a5f1939e
* :yellow_heart: [Build #342959](https://buildkite.com/elastic/kibana-pull-request/builds/342959) was flaky 5809d6fe56714f3b07f7799cd06055e43285725a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346548","number":346548,"commit":"73f6e0027a76593fe8f77fb9ce2adb16e3e94761"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346379","number":346379,"commit":"ca92182ba2d876ceace2de8ea55e3d5d5d106efc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346335","number":346335,"commit":"55265594a899376fa99d738cc53c36bef56da794"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346136","number":346136,"commit":"8eb8d7b8f06d132d3a5bdf6255e1adb873f10584"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345562","number":345562,"commit":"a3239b6f06d0016ff6c2bea1d5abf821a5f1939e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342959","number":342959,"commit":"5809d6fe56714f3b07f7799cd06055e43285725a"}],"number":346548}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346560)
* Commit: 73f6e0027a76593fe8f77fb9ce2adb16e3e94761

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/346560#0199a3e9-107b-4273-a859-dd6e593f83ad)

### History
* :broken_heart: [Build #346548](https://buildkite.com/elastic/kibana-pull-request/builds/346548) failed 73f6e0027a76593fe8f77fb9ce2adb16e3e94761
* :broken_heart: [Build #346379](https://buildkite.com/elastic/kibana-pull-request/builds/346379) failed ca92182ba2d876ceace2de8ea55e3d5d5d106efc
* :broken_heart: [Build #346335](https://buildkite.com/elastic/kibana-pull-request/builds/346335) failed 55265594a899376fa99d738cc53c36bef56da794
* :broken_heart: [Build #346136](https://buildkite.com/elastic/kibana-pull-request/builds/346136) failed 8eb8d7b8f06d132d3a5bdf6255e1adb873f10584
* :broken_heart: [Build #345562](https://buildkite.com/elastic/kibana-pull-request/builds/345562) failed a3239b6f06d0016ff6c2bea1d5abf821a5f1939e
* :yellow_heart: [Build #342959](https://buildkite.com/elastic/kibana-pull-request/builds/342959) was flaky 5809d6fe56714f3b07f7799cd06055e43285725a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346560","number":346560,"commit":"73f6e0027a76593fe8f77fb9ce2adb16e3e94761"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346548","number":346548,"commit":"73f6e0027a76593fe8f77fb9ce2adb16e3e94761"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346379","number":346379,"commit":"ca92182ba2d876ceace2de8ea55e3d5d5d106efc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346335","number":346335,"commit":"55265594a899376fa99d738cc53c36bef56da794"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346136","number":346136,"commit":"8eb8d7b8f06d132d3a5bdf6255e1adb873f10584"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345562","number":345562,"commit":"a3239b6f06d0016ff6c2bea1d5abf821a5f1939e"}],"number":346560}
buildkite-pr-comment-->

VladimirFilonov committed:
Refactor Workflow List and Detail Page for Improved Execution Handling

- Simplified the workflow execution logic in WorkflowList by directly setting the workflow item for execution.
- Removed the unused test workflow modal state and related logic from WorkflowDetailPage to streamline the component.
- Updated the handleRunClick function to directly open the execution modal without additional parameters.

These changes enhance the clarity and efficiency of the workflow execution process in the UI.

VladimirFilonov committed:
Remove unused runWorkflow function from WorkflowDetailPage to streamline workflow actions. This change enhances the clarity and efficiency of the component by eliminating unnecessary code.

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

Kiryous reviewed (commented):
Approved with a few small comments

semd commented on src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx:
```diff
@@ -0,0 +1,385 @@
+import { take } from 'rxjs';
+import { useKibana } from '../../../hooks/use_kibana';
+
+interface Document {
+  _id: string;
+  _index: string;
+  _source: {
+    '@timestamp': string;
+    agent?: string;
+    user?: string;
+    [key: string]: any;
+  };
+}
```
The `Document` should only define the `_source` shape. For the rest, we can use `SearchHit<Document>`.

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

# Conflicts:
#	src/platform/plugins/shared/workflows_management/public/pages/workflow_detail/ui/workflow_detail_page.tsx
#	src/platform/plugins/shared/workflows_management/tsconfig.json

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feat/improve-run-workflow-modal

VladimirFilonov committed:
Remove unused event handlers and optimize JSON editor settings in workflow execution forms

VladimirFilonov committed:
Add comment why we need global CSS to prevent modal overlay on SearchBar autocomplete

VladimirFilonov committed:
Remove unused rule property from alert object in WorkflowExecuteEventForm component

VladimirFilonov committed:
Refactor WorkflowExecuteIndexForm to use SearchHit<Document> type for documents

VladimirFilonov commented on src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx:
```diff
@@ -0,0 +1,385 @@
+        },
+      };
+
+      const request = {
+        params: {
+          index: selectedDataView.getIndexPattern(),
+          body: {
+            query: searchQuery,
+            size: 50,
+            sort: [{ '@timestamp': { order: 'desc' } }],
+            _source: ['@timestamp', 'agent', 'user', 'message', 'host.name', 'source.ip'],
+          },
+        },
+      };
+
+      const response = await services.data.search.search(request).pipe(take(1)).toPromise();
+
+      if (
+        response &&
+        response.rawResponse &&
+        response.rawResponse.hits &&
+        response.rawResponse.hits.hits
+      ) {
+        setDocuments(response.rawResponse.hits.hits as Document[]);
+      } else {
+        setDocuments([]);
+      }
+    } catch (err) {
+      setErrors(err instanceof Error ? err.message : 'Failed to fetch documents');
+      setDocuments([]);
+    } finally {
+      setDocumentsLoading(false);
+    }
```
Wow, that's great - thank you for highlighting this possibility!

VladimirFilonov committed:
Refactor WorkflowExecuteIndexForm to streamline document fetching and remove unused currentUser state

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

VladimirFilonov committed:
Remove unused getCurrentUser function from WorkflowExecuteIndexForm to streamline code and improve maintainability

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/348043)
* Commit: 7b2d0b92cfecff53696c17f9d08b30136e7c05a9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236173)

### Failed CI Steps
* [FTR Configs #27](https://buildkite.com/elastic/kibana-pull-request/builds/348043#0199bd50-d65e-4259-9f16-39c8848130fd)
* [FTR Configs #31](https://buildkite.com/elastic/kibana-pull-request/builds/348043#0199bd50-d665-4ef4-a4a3-e8931b004ed1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/348043#0199bd50-d665-4ef4-a4a3-e8931b004ed1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/348043/jobs/0199bd50-d665-4ef4-a4a3-e8931b004ed1/artifacts/0199bd77-55bc-4218-8107-d7869e203202) FTR Configs #<span></span>31 / endpoint Endpoint permissions: when running with user/role [hunter_no_actions] should display endpoint data on Host Details
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/348043#0199bd50-d65e-4259-9f16-39c8848130fd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/348043/jobs/0199bd50-d65e-4259-9f16-39c8848130fd/artifacts/0199bd76-19bf-470f-b24b-f46bc6fe86da) FTR Configs #<span></span>27 / logstash pipeline list route add button links to the empty pipeline editor

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/09552c79f4c574ead6e9a83a5ea816a98b0e38c2) | [after](https://github.com/elastic/kibana/commit/7b2d0b92cfecff53696c17f9d08b30136e7c05a9) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 647 | 648 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/09552c79f4c574ead6e9a83a5ea816a98b0e38c2) | [after](https://github.com/elastic/kibana/commit/7b2d0b92cfecff53696c17f9d08b30136e7c05a9) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 2.0MB | 2.0MB | +2.1KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/09552c79f4c574ead6e9a83a5ea816a98b0e38c2) | [after](https://github.com/elastic/kibana/commit/7b2d0b92cfecff53696c17f9d08b30136e7c05a9) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 6.0KB | 6.2KB | +197.0B |

<details><summary>Unknown metric groups</summary>

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/09552c79f4c574ead6e9a83a5ea816a98b0e38c2) | [after](https://github.com/elastic/kibana/commit/7b2d0b92cfecff53696c17f9d08b30136e7c05a9) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 0 | 2 | +2 |


</details>


### History
* :broken_heart: [Build #347863](https://buildkite.com/elastic/kibana-pull-request/builds/347863) failed 3e8ec7371e851796f8958eecbb5325768849239c
* :yellow_heart: [Build #346738](https://buildkite.com/elastic/kibana-pull-request/builds/346738) was flaky 61c5ddfea32ddb5ff9263cb662269d31317364b6
* :broken_heart: [Build #346599](https://buildkite.com/elastic/kibana-pull-request/builds/346599) failed 36a1f01dd2e85ee4d1de1233314dda54a1c5338b
* :broken_heart: [Build #346560](https://buildkite.com/elastic/kibana-pull-request/builds/346560) failed 73f6e0027a76593fe8f77fb9ce2adb16e3e94761
* :broken_heart: [Build #346548](https://buildkite.com/elastic/kibana-pull-request/builds/346548) failed 73f6e0027a76593fe8f77fb9ce2adb16e3e94761
* :broken_heart: [Build #346379](https://buildkite.com/elastic/kibana-pull-request/builds/346379) failed ca92182ba2d876ceace2de8ea55e3d5d5d106efc

cc @VladimirFilonov

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348043","number":348043,"commit":"7b2d0b92cfecff53696c17f9d08b30136e7c05a9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347863","number":347863,"commit":"3e8ec7371e851796f8958eecbb5325768849239c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346738","number":346738,"commit":"61c5ddfea32ddb5ff9263cb662269d31317364b6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346599","number":346599,"commit":"36a1f01dd2e85ee4d1de1233314dda54a1c5338b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346560","number":346560,"commit":"73f6e0027a76593fe8f77fb9ce2adb16e3e94761"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346548","number":346548,"commit":"73f6e0027a76593fe8f77fb9ce2adb16e3e94761"}],"number":348043}
buildkite-pr-comment-->
